eth2/state_processing/src/per_epoch_processing/registry_updates.rs

```diff
@@ -15,29 +15,24 @@ pub fn process_registry_updates<T: EthSpec>(
     // We assume it's safe to re-order the change in eligibility and `initiate_validator_exit`.
     // Rest assured exiting validators will still be exited in the same order as in the spec.
     let current_epoch = state.current_epoch();
-    let is_exiting_validator = |validator: &Validator| {
+    let is_ejectable = |validator: &Validator| {
         validator.is_active_at(current_epoch)
             && validator.effective_balance <= spec.ejection_balance
     };
-    let (eligible_validators, exiting_validators): (Vec<_>, Vec<_>) = state
+    let indices_to_update: Vec<_> = state
         .validators
         .iter()
         .enumerate()
         .filter(|(_, validator)| {
-            validator.is_eligible_for_activation_queue(spec) || is_exiting_validator(validator)
+            validator.is_eligible_for_activation_queue(spec) || is_ejectable(validator)
         })
-        .partition_map(|(index, validator)| {
-            if validator.is_eligible_for_activation_queue(spec) {
-                Either::Left(index)
-            } else {
-                Either::Right(index)
-            }
-        });
-    for index in eligible_validators {
-        state.validators[index].activation_eligibility_epoch = current_epoch + 1;
-    }
-    for index in exiting_validators {
-        initiate_validator_exit(state, index, spec)?;
+    for index in indices_to_update {
+        if state.validators[index].is_eligible_for_activation_queue(spec) {
+            state.validators[index].activation_eligibility_epoch = current_epoch + 1;
+        }
+        if is_ejectable(validator) {
```

Comment:
The `validator` of type `&Validator` doesn't exist here, because we're only iterating over a list of indices.